### PR TITLE
Edit a Reminder/Note 

### PR DESCRIPTION
**Pull request by chelletuerk**
## Purpose

Closes issue #10

Allow user to edit existing reminder.  Allow user to save new edited reminder and upon clicking save the user returns to the route showing the new updated reminder.

## Approach

User has the ability to update existing reminder.

### Learning
- [Building a Todo app using Ember.js & Firebase – Rails, Ember & Beyond](https://medium.com/rails-ember-beyond/building-a-todo-app-using-ember-js-firebase-ba677e81e561#.cpun552sg)
- [Transitioning routes](http://stackoverflow.com/questions/11552417/emberjs-how-to-transition-to-a-router-from-a-controllers-action)
- [Ember Tutorial](https://github.com/ssriera/ember-to-do-list/wiki/Tutorial)
- [Ember Website Acceptance Test](http://guides.emberjs.com/v1.12.0/testing/acceptance/)
- [StackOverflow routing](http://stackoverflow.com/questions/11552417/emberjs-how-to-transition-to-a-router-from-a-controllers-action)
- [`objects.pop` is not a function from transitionTo #12719](https://github.com/emberjs/ember.js/issues/12719)

### Open Questions and Pre-Merge TODOs

- [x] When viewing an existing reminder, the user can click on an “Edit” button where they can modify the title, notes, or date of the reminder.
- [x] When editing a note, there is a “Save” button that commits the changes to the server.
- [x] When the user hits the save button, they are returned to the original individual note path (ie: reminders/1).

### Test coverage 

Wrote acceptance tests for redirect after editing a reminder. 

#### Before
screen capture on 2017-02-14 at 18-18-56

#### After

screen capture on 2017-02-14 at 18-18-56

### CODE REVIEW

File: app/components/edit-reminder.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+
+  store: Ember.inject.service(),
+  router: Ember.inject.service('-routing'),
```
Comment (Tman22) [inline app/components/edit-reminder.js]: This is interesting. I'm not sure how I feel about this. I believe re-routing should be passed to the controller or preferably the router.

File: app/router.js [hunk 2]
```diff
@@ -10,6 +10,7 @@ Router.map(function() {
   this.route('reminders', function() {
     this.route('reminder', { path: '/:reminder_id' });
```
Comment (Tman22) [inline app/router.js]: Can we wrap edit inside of this?

File: tests/acceptance/reminders/edit-test.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { test } from 'qunit';
+import moduleForAcceptance from 'remember/tests/helpers/module-for-acceptance';
+
+moduleForAcceptance('Acceptance | reminders/edit');
+
+test('editing a reminder returns user to reminder item', function(assert) {
+  server.createList('reminder', 5);
+
```
Comment (Tman22) [inline tests/acceptance/reminders/edit-test.js]: Can we create a more thorough test?

Comment (Tman22) [inline tests/acceptance/reminders/edit-test.js]: Can we create a more thorough test?

File: app/components/edit-reminder.js (hunk 1 above)
Comment (chelletuerk) [reply app/components/edit-reminder.js]: Both Eric and I agree that it IS interesting.  We had a heck of a time trying to route back to /reminders/id/.  This one the only (of many different attempted ways) we could get it to work.  We tried with a controller and had no success.  We can certainly take another look, though.

**Comment by esayler** [comment]
@martensonbj @Tman22 we moved the redirect on edit button action to the edit controller, but kept the routes unnested for now.

File: app/router.js (hunk 2 above)
Comment (Tman22) [reply app/router.js]: Hey so you still need to do this.

File: tests/acceptance/reminders/edit-test.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+test('editing a reminder title changes its title in list', function(assert) {
+  server.createList('reminder', 5);
+
+  visit('/reminders/edit/1');
+
+  andThen(function() {
+    fillIn('.reminder-title', 'Hello World');
+    click('.btn-update');
+
```
Comment (Tman22) [inline tests/acceptance/reminders/edit-test.js]: Spacing, also the fillIn, and click don't need to be in a andThen()

Comment (esayler) [inline tests/acceptance/reminders/edit-test.js]: Whaaaaat do you mean by 'do this'?

File: app/router.js (hunk 2 above)
Comment (esayler) [reply app/router.js]: Oh, btw, this is chelle...we got this puppy going without wrapping edit.  We did what you said!  Please merge.   @Tman22   😆

File: tests/acceptance/reminders/edit-test.js (hunk 4 above)
Comment (esayler) [reply tests/acceptance/reminders/edit-test.js]: Ok we can try to change the route...it might take significant refactoring(?)

**Comment by esayler** [comment]
@Tman22 ok we nested the edit route. Thanks for the help!

**Pull request merged**